superset/db_engine_specs/base.py

```diff
@@ -80,25 +81,25 @@ class TimeGrain(NamedTuple):
 
 
 builtin_time_grains: dict[str | None, str] = {
-    "PT1S": __("Second"),
```

Comment:
The type declaration on line 82 should be updated; the key should be `TimeGrainConstants`, right?

Reply:
It's actually not the class but the string values defined in it

Reply:
Oh right - I believe it should in fact be an enum (I'll comment on the relevant line)

Reply:
If you make the `Enum` change above, you should be able to change the above to 
```
builtin_time_grains: dict[Optional[TimeGrainConstants], str] = {
```

Reply:
Check this [comment](https://github.com/apache/superset/pull/24176#discussion_r1214443954)